pkg/cloudprovider/providers/openstack/openstack_routes.go

```diff
@@ -276,3 +286,44 @@ func (r *Routes) DeleteRoute(clusterName string, route *cloudprovider.Route) err
 	onFailure.Disarm()
 	return nil
 }
+
+func getPortIDByIP(compute *gophercloud.ServiceClient, targetNode types.NodeName, ipAddress string) (string, error) {
+	srv, err := getServerByName(compute, targetNode)
+	if err != nil {
+		return "", err
+	}
+
+	interfaces, err := getAttachedInterfacesByID(compute, srv.ID)
+	if err != nil {
+		return "", err
+	}
+
+	var portID string
+	for _, intf := range interfaces {
+		for _, fixedIP := range intf.FixedIPs {
+			if fixedIP.IPAddress == ipAddress {
+				portID = intf.PortID
+				break
+			}
+		}
+	}
+
+	if portID == "" {
```

Comment:
If we do the above, we can skip the if condition here and just `return "", ErrNotFound` (and remove line 315 too)